src/storage/page_buffer.c

```diff
@@ -14718,6 +14722,30 @@ pgbuf_rv_dealloc_redo (THREAD_ENTRY * thread_p, const LOG_RCV * rcv)
   pgbuf_set_page_ptype (thread_p, rcv->pgptr, PAGE_UNKNOWN);
   pgbuf_set_tde_algorithm (thread_p, rcv->pgptr, TDE_ALGORITHM_NONE, true);
   pgbuf_set_dirty (thread_p, rcv->pgptr, DONT_FREE);
+
+#if defined(SERVER_MODE)
+  if (is_passive_transaction_server ())
+    {
+      /* on a passive transaction server; upon de-allocation, remove the page
+       * from page buffer altogether; needed because to provide a fail-safe mechanism
+       * for page re-allocation: when the same page is re-allocated in a subsequent
+       * transaction, the replication mechanism will fail to fix an 'PAGE_UNKNOWN' page */
+#if !defined(NDEBUG)
+      /* crash if there are other, client, transactions waiting in line to fix the same page;
+       * architecturally, this should not be the case (as in a "monolithic" server as well) */
+      PGBUF_BCB *bufptr;
+      CAST_PGPTR_TO_BFPTR (bufptr, rcv->pgptr);
+      assert (bufptr->next_wait_thrd == nullptr);
+#endif
+      const int error_code = pgbuf_invalidate (thread_p, rcv->pgptr);
```

Comment:
I'm not sure `pgbuf_invalidate ()` is safe function to use while transactions are running on PTS.

Can we be sure that no transaction has fixed the page in the PTS at the time the pgbuf_rv_dealloc_redo function is called?
I think it would be fine if the pgbuf_rv_dealloc_redo () was a replication that occurred during the vacuum process, but I am not sure if there are other cases.